Zend/zend_alloc.c

```diff
@@ -2957,7 +2980,9 @@ ZEND_API bool zend_mm_is_custom_heap(zend_mm_heap *new_heap)
 ZEND_API void zend_mm_set_custom_handlers(zend_mm_heap *heap,
                                           void* (*_malloc)(size_t ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC),
                                           void  (*_free)(void* ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC),
-                                          void* (*_realloc)(void*, size_t ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC))
+                                          void* (*_realloc)(void*, size_t ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC),
+                                          size_t (*_gc)(void),
+                                          void   (*_shutdown)(bool, bool))
```

Comment:
May be it's better to keep the existing function prototypes and add ``zend_mm_set_custom_handlers_ex()``.
Anyway, this shouldn't make big problems for third-party API users.

Reply:
Would the `zend_mm_set_custom_handlers_ex()` function you are thinking of only get the `gc` and `shutdown` function pointers or all?
I was suggesting this way, as I do want to avoid synchronisation issues between `zend_mm_set_custom_handlers()` and zend_mm_set_custom_handlers_ex()`. Also `gc` and `shutdown` are allowed to be `NULL`, so making extension compatible should be easy.

Reply:
All pointers would be the most convenient, and would solve the synchronization issue, as well as the issue you mention in the PR description.

`zend_mm_set_custom_handlers()` could then be implemented as a call to `zend_mm_set_custom_handlers_ex(malloc, free, realloc, NULL, NULL)`.

Reply:
That makes sense, thanks for the hint. I implemented those changes.